rocm_getpath.cpp

```diff
@@ -25,10 +25,13 @@
 
 #include <string.h>
 #include <stdlib.h>
-#include <limits.h> /* PATH_MAX */
 #include <stdio.h>
+#if defined(_WIN32) || defined(__CYGWIN__)
+// No Windows-specific includes.
+#else
 #include <link.h>
 #include <dlfcn.h>
+#endif
```

Comment:
dlfcn.h header is included only for linux. But these checks are not added for dlopen functions.

Reply:
I see one use of `dlopen` in this repository (https://github.com/search?q=repo%3AROCm%2Frocm-core+dlopen&type=code), which is already gated on `BUILD_SHARED_LIBS`: https://github.com/ROCm/rocm-core/blob/75be79f4708a419d3755500ca67ed9ec062e15bf/rocm_getpath.cpp#L127-L129

In TheRock we set `-DBUILD_SHARED_LIBS` on Windows to avoid that code path: https://github.com/ROCm/TheRock/blob/2f6924203448ec81044eb665276a9ebe841f871d/base/CMakeLists.txt#L28-L39

Reply:
We cannot have the code which take care of only ROCK use case. If anyone builds in windows without BUILD_SHARED_LIBS properly set can get error. Since the support for windows is getting added lets do it everywhere properly. I dont have a windows dev env else I could have helped.

Reply:
Would an error like this be helpful?

```c++
  // If Environment Variable is not set
  // use dl APIs to get target lib path
  // and get rocm base install path using the lib Path.
#if BUILD_SHARED_LIBS
#if defined(_WIN32) || defined(__CYGWIN__)
  #error BUILD_SHARED_LIBS is not supported on Windows yet
#endif
  sprintf(libFileName, "lib%s.so", TARGET_LIBRARY_NAME);
  void *handle=dlopen(libFileName,RTLD_NOW);
  // ...
```

I'm not sure yet what would be involved in enabling shared library support here and in other ROCm components on Windows.

Reply:
Looks fine for me. 
@frepaul do you have any concerns for having #error

Reply:
if  "BUILD_SHARED_LIBS"  is not used in windows does that mean that getROCmInstallPath is not going to fetch anything other than ROCM_PATH that could be again only on Linux. In effect this API getROCmInstallPath is not supported in windows I guess?